echidna/scripts/zero_nu_limit.py

```diff
@@ -23,56 +23,80 @@
     Te130_0n2b = store.load(echidna.__echidna_home__ +
-                            "data/TeLoadedTe130_0n2b.ntuple_reco.hdf5")
+                            "/data/Te130_0n2b_mc_smeared.hdf5")
+    print Te130_0n2b._name
 
     # Create background spectra
     Te130_2n2b = store.load(echidna.__echidna_home__ +
-                            "data/TeLoadedTe130_2n2b.ntuple_reco.hdf5")
-    B8 = store.load(echidna.__echidna_home__ +
-                    "data/TeLoadedB8.ntuple_reco.hdf5")
+                            "/data/Te130_2n2b_mc_smeared.hdf5")
+    print Te130_2n2b._name
+    B8_Solar = store.load(echidna.__echidna_home__ +
+                          "/data/B8_Solar_mc_smeared.hdf5")
+    print B8_Solar._name
 
     # Shrink spectra to 5 years - livetime used by Andy
     # And make 3.5m fiducial volume cut
+    # Temporary fix to _num_decays and _raw_events
+    Te130_0n2b._num_decays = Te130_0n2b.sum()
+    Te130_0n2b._raw_events = 200034
     Te130_0n2b.shrink(0.0, 10.0, 0.0, 3500.0, 0.0, 5.0)
+    Te130_2n2b._num_decays = Te130_2n2b.sum()
+    Te130_2n2b._raw_events = 75073953
     Te130_2n2b.shrink(0.0, 10.0, 0.0, 3500.0, 0.0, 5.0)
-    B8.shrink(0.0, 10.0, 0.0, 3500.0, 0.0, 5.0)
+    B8_Solar._num_decays = B8_Solar.sum()
+    B8_Solar._raw_events = 106228
+    B8_Solar.shrink(0.0, 10.0, 0.0, 3500.0, 0.0, 5.0)
 
     # Create list of backgrounds
     backgrounds = []
     backgrounds.append(Te130_2n2b)
-    backgrounds.append(B8)
+    backgrounds.append(B8_Solar)
 
     # Initialise limit setting class
     roi = (2.46, 2.68)  # Define ROI - as used by Andy
-    set_limit = limit_setting.LimitSetting(Te130_0n2b, backgrounds, roi=roi)
+    set_limit = limit_setting.LimitSetting(Te130_0n2b, backgrounds, roi=roi,
+                                           pre_shrink=True,
+                                           verbose=args.verbose)
 
     # Configure Te130_0n2b
-    Te130_0n2b_counts = numpy.arange(10.0, 6.0e3, 10.0, dtype=float)
-    Te130_0n2b_prior = 4.2077507e3  # Based on T_1/2 = 6.2e24 y for 10 years
+    Te130_0n2b_counts = numpy.arange(5.0, 500.0, 5.0, dtype=float)
+    Te130_0n2b_prior = 262.0143  # Based on T_1/2 = 9.94e25 y @ 90% CL
+                                 # (SNO+-doc-2593-v8) for 5 year livetime
+                                 # Note extrapolating here to 10 years
     Te130_0n2b_config = limit_config.LimitConfig(Te130_0n2b_prior,
                                                  Te130_0n2b_counts)
     set_limit.configure_signal(Te130_0n2b_config)
 
     # Configure Te130_2n2b
-    Te130_2n2b_counts = numpy.arange(11.342632e6, 11.342633e6,
+    Te130_2n2b_counts = numpy.arange(11.323579e6, 11.323580e6,
                                      1.0, dtype=float)
     # no penalty term to start with so just an array containing one value
-    Te130_2n2b_prior = 11.342632e6  # Based on T_1/2 = 2.3e21 y for 10 years
+    Te130_2n2b_prior = 11.323579e6  # Based on T_1/2 = 2.3e21 y for 10 years
```

Comment:
This is probably one of the next big things we need to think about. Passing by argument is fine for a signal and one or two background, but after that it is going to become impractical to be passing every hdf5 file and all relevant parameters via the command line. We also don't want to hard code all the scaling information for every background and have it on github, as it is potentially sensitive information and the repository is public. I think this may be where were going to start needing (I hate to say it) a ***database***! I did speak briefly with @mjmottram about databases for storing hdf5 files, would it work for storing this sort of information too?

However I think all of this is getting a little beyond the scope of this script. It was intended as just a simple example of what the limit setting can do, that anyone could run, without too much knowledge of echidna and the limit setting routines and just set a basic limit.